### PR TITLE
[Root Objects] Order by specified priority

### PR DESCRIPTION
**Pull request by jvigliotta**
### All Submissions:

Have added a priority argument to objectAPI's `addRoot`mehtod. Updated tests and added documentation, backwards compatible with previous array and async function support.

Have added a default priority for "My Items" to "LOW". This is related to the need for priority support for root objects in the first place.

Also added docs for Priority API since it's used in the example for addRoot.

closes: #4638

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue?
* [x] Documentation added?

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -247,16 +249,24 @@ To do so, use the `addRoot` method of the object API.
...
-    });
+    namespace: "example.namespace",
+    key: "my-key"
+},
+openmct.priority.HIGH);
 ```
 
-The `addRoot` function takes a single [object identifier](#domain-objects-and-identifiers) 
-as an argument. 
+The `addRoot` function takes a two arguments, the first can be an [object identifier](#domain-objects-and-identifiers) for a root level object, or an array of identifiers for root
```
Comment (akhenry) [inline API.md:258]: Awesome, I love these API docs, nice one!

File: API.md [hunk 2]
```diff
@@ -1051,3 +1061,25 @@ A completely custom indicator can be added by simply providing a DOM element to
         element: domNode
     });
 ```
+
+## Priority API
```
Comment (akhenry) [inline API.md:1065]: Thanks for filling in this missing API documentation

**Review by akhenry — APPROVED** [review]
Looks great, thanks for sorting this out!

### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
5. Has associated issue been labelled `unverified`? (only applicable if this PR closes the issue) Y

File: API.md (hunk 1 above)
Comment (jvigliotta) [reply API.md:258]: Thanks! going to try to keep this up as I go along.

Comment (jvigliotta) [reply API.md:258]: Oh dang... "takes a two arguments", I guess if we assume it's Mario speaking, we'll be ok.

**Pull request merged**